src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadServiceTest.kt

```diff
@@ -0,0 +1,1306 @@
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.GroupedByCom
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.LicenceSummary
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.Prisoner
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.ProbationPractitioner
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceQueryObject
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.CaseloadService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.LicenceService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.TestData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerHdcStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.CommunityApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.OffenderDetail
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.OffenderManager
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.OtherIds
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.ProbationSearchApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.StaffDetail
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.StaffHuman
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.User
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.CaViewCasesTab
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceKind
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import java.time.Clock
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.ZoneId
+
+class CaCaseloadServiceTest {
+  private val caseloadService = mock<CaseloadService>()
+  private val probationSearchApiClient = mock<ProbationSearchApiClient>()
+  private val licenceService = mock<LicenceService>()
+  private val prisonApiClient = mock<PrisonApiClient>()
+  private val communityApiClient = mock<CommunityApiClient>()
+
+  private val service = CaCaseloadService(
+    caseloadService,
+    clock,
+    probationSearchApiClient,
+    licenceService,
+    prisonApiClient,
+    communityApiClient,
+  )
+
+  private val statuses = listOf(
+    LicenceStatus.APPROVED,
+    LicenceStatus.SUBMITTED,
+    LicenceStatus.IN_PROGRESS,
+    LicenceStatus.TIMED_OUT,
+    LicenceStatus.ACTIVE,
+  )
+
+  private val licenceQueryObject = LicenceQueryObject(
+    statusCodes = statuses,
+    prisonCodes = listOf("BAI"),
+//    sortBy = "conditionalReleaseDate",
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏needs removing if not used.

Reply:
un commented and using it.